src/frontend/src/catalog/pg_catalog/mod.rs

```diff
@@ -123,6 +126,103 @@ impl SysCatalogReaderImpl {
             .collect_vec())
     }
 
+    fn read_class_info(&self) -> Result<Vec<Row>> {
+        let reader = self.catalog_reader.read_guard();
+        let schemas = reader.iter_schemas(&self.auth_context.database)?;
+        let schema_infos = reader.get_all_schema_info(&self.auth_context.database)?;
+
+        Ok(schemas
+            .zip_eq(schema_infos.iter())
+            .flat_map(|(schema, schema_info)| {
+                let rows = schema
+                    .iter_table()
+                    .map(|table| {
+                        Row::new(vec![
+                            Some(ScalarImpl::Int32(table.id.table_id() as i32)),
+                            Some(ScalarImpl::Utf8(table.name.clone())),
+                            Some(ScalarImpl::Int32(schema_info.id as i32)),
+                            Some(ScalarImpl::Int32(table.owner as i32)),
+                            Some(ScalarImpl::Utf8("table".to_string())),
+                        ])
+                    })
+                    .collect_vec();
+
+                let mvs = schema
+                    .iter_mv()
+                    .map(|mv| {
+                        Row::new(vec![
+                            Some(ScalarImpl::Int32(mv.id.table_id() as i32)),
+                            Some(ScalarImpl::Utf8(mv.name.clone())),
+                            Some(ScalarImpl::Int32(schema_info.id as i32)),
+                            Some(ScalarImpl::Int32(mv.owner as i32)),
+                            Some(ScalarImpl::Utf8("materialized view".to_string())),
+                        ])
+                    })
+                    .collect_vec();
+
+                let indexes = schema
+                    .iter_index()
+                    .map(|mv| {
+                        Row::new(vec![
+                            Some(ScalarImpl::Int32(mv.id.table_id() as i32)),
+                            Some(ScalarImpl::Utf8(mv.name.clone())),
+                            Some(ScalarImpl::Int32(schema_info.id as i32)),
+                            Some(ScalarImpl::Int32(mv.owner as i32)),
+                            Some(ScalarImpl::Utf8("index".to_string())),
+                        ])
+                    })
+                    .collect_vec();
+
+                let sources = schema
+                    .iter_source()
+                    .map(|source| {
+                        Row::new(vec![
+                            Some(ScalarImpl::Int32(source.id as i32)),
+                            Some(ScalarImpl::Utf8(source.name.clone())),
+                            Some(ScalarImpl::Int32(schema_info.id as i32)),
+                            Some(ScalarImpl::Int32(source.owner as i32)),
+                            Some(ScalarImpl::Utf8("source".to_string())),
+                        ])
+                    })
+                    .collect_vec();
+
+                let msources = schema
+                    .iter_materialized_source()
```

Comment:
`materialized source` is subset of `source`, I think we can only provide `source` in `pg_class`.